tls/s2n_client_hello.c

```diff
@@ -368,15 +368,15 @@ int s2n_client_hello_send(struct s2n_connection *conn)
 
     struct s2n_stuffer *out = &conn->handshake.io;
     struct s2n_stuffer client_random = {0};
-    struct s2n_blob b, r = {0};
-    uint8_t client_protocol_version[S2N_TLS_PROTOCOL_VERSION_LEN];
+    uint8_t client_protocol_version[S2N_TLS_PROTOCOL_VERSION_LEN] = {0};
 
+    struct s2n_blob b = {0};
     GUARD(s2n_blob_init(&b, conn->secure.client_random, S2N_TLS_RANDOM_DATA_LEN));
     /* Create the client random data */
     GUARD(s2n_stuffer_init(&client_random, &b));
 
-    r.data = s2n_stuffer_raw_write(&client_random, S2N_TLS_RANDOM_DATA_LEN);
-    r.size = S2N_TLS_RANDOM_DATA_LEN;
+    struct s2n_blob r = {0};
+    GUARD(s2n_blob_init(&r, s2n_stuffer_raw_write(&client_random, S2N_TLS_RANDOM_DATA_LEN), S2N_TLS_RANDOM_DATA_LEN));
```

Comment:
Can we use `s2n_stack_blob()` instead? It internally calls `s2n_blob_init()`. If we want the blob to point to a memory address other than the stack maybe we can add a new `s2n_raw_blob()` Macro?

https://github.com/awslabs/s2n/blob/master/utils/s2n_blob.h#L53

Reply:
Yes, that looks to be exactly what we should do!

Reply:
Is there a reason `s2n_stack_blob` is not in all caps?

Reply:
I'm going to do this change in a followup PR